bundle/src/bundler.rs

```diff
@@ -19,6 +26,31 @@ impl BundlerUtil {
         Self { meta }
     }
 
+    /// Reads and decompresses a .tar.zstd file from an input stream into just a `meta.json` file
+    ///
+    pub async fn parse_meta_from_tarball<R: AsyncBufRead>(input: R) -> anyhow::Result<Self> {
```

Comment:
I've also considering adding more fields to `BundlerUtil`, including the compressed/decompressed sizes. Lmk others' thoughts. I can handle as a follow-up if desired